src/components/ContactsList/index.js

```diff
@@ -33,31 +56,53 @@ export const withContacts = BaseComponent =>
     };
 
     async componentDidMount() {
-      const contactsRaw = await cozy.client
-        .fetchJSON("GET", "/data/io.cozy.contacts/_all_docs?include_docs=true")
-        .then(data => data.rows)
-        .catch(error => {
-          this.setState(state => ({ ...state, contacts: [], error }));
-        });
-      const contacts = (contactsRaw || [])
-        .map(row => ({
-          ...row.doc
-        }))
-        .filter(contact => contact.email);
+      const contacts = await cozyHTTP.getContacts();
       this.setState(state => ({
         ...state,
         contacts: [...state.contacts, ...contacts],
         loading: false
       }));
     }
 
+    deleteContact = async contact => {
+      const index = this.state.contacts.indexOf(contact);
+      this.setState(state => ({
+        ...state,
+        contacts: [
+          ...state.contacts.slice(0, index),
+          ...state.contacts.slice(index + 1)
+        ]
+      }));
+      try {
+        const result = await cozyHTTP.deleteContact(contact);
+        if (!result.deleted)
+          throw new Error("Contact not deleted on the stack");
+      } catch (ex) {
+        this.setState(state => ({
+          ...state,
+          contacts: [
+            ...state.contacts.slice(0, index),
+            contact,
+            ...state.contacts.slice(index + 1)
+          ]
+        }));
+        throw ex;
+      }
```

Comment:
As I deleted contact on UI before the stack, if the stack could not delete (HTTP response, exception, etc.) the contact, we readd it to the UI.

Reply:
:+1: for the optimistic delete ;)

Reply:
Ah ouais, voilà, c'est comme ça que ça s'appelle. ;-)
Avec React, c'est tellement simple à mette en place !